x-pack/plugins/enterprise_search/public/applications/app_search/components/relevance_tuning/types.ts

```diff
@@ -7,17 +7,30 @@
 
 export type BoostType = 'value' | 'functional' | 'proximity';
 
-export interface Boost {
-  type: BoostType;
+export interface BaseBoost {
   operation?: string;
   function?: string;
+}
+
+export interface Boost extends BaseBoost {
+  type: BoostType;
   newBoost?: boolean;
   center?: string | number;
-  value?: string | number | string[] | number[];
+  value?: Array<string | number>;
```

Comment:
Can you elaborate on some of the expected values you'd see for this `value` property. I'm trying to understand why you moved from `string[] | number[];` to `Array<string | number>;` which I think is technically different right? Are you expecting `["1.5", 2, 3, "4"]`?

Reply:
You know, based one what can be set with just the UI, it actually appeared to me initially that it was ONLY arrays of strings that were valid.

But then I opened up the API... and it's actually super loose:
```
[
      {
        "type": "value",
        "factor": 9.5,
        "value": 1
      },
      {
        "type": "value",
        "factor": 9.5,
        "value": "1"
      },
      {
        "type": "value",
        "factor": 9.5,
        "value": [1]
      },
      {
        "type": "value",
        "factor": 9.5,
        "value": ["1"]
      },
      {
        "type": "value",
        "factor": 9.5,
        "value": ["1", 1, "2", 2]
      }
    ]
```

So the type I have here *is* more correct that `string[] | number[]` since they can be mixed, but it looks like I need to expand it to:
```
 value?: string | number | Array<string | number>;
```

I actually may put something right up front at initialization that wraps `string` and `number` in an array so I can keep the type as is. Will follow up shortly.

Reply:
Oh my gosh, the API accepts *literally* anything:
```
      {
        "type": "value",
        "factor": 9.5,
        "value": ["1", 1, "2", 2, true, { "b" :"a"}]
      }
```

Reply:
Oof, no good suggestions here, just a gif reaction and also a thank you for clarifying the typing. I think we should draw the line at allowing non string/nums in our client-side code at least 😬 

<img src="https://media.giphy.com/media/kexK6MPhNziEvkaX3u/giphy.gif" alt="yikes" />

Reply:
I made the following change to address this: f47749c

Our UI *already* only sends string values to the server. Because of this, there's no real reason to deal with any other types in the UI. So for simplicity, I am calling `toString` on all values as soon as we fetch them from the server.

Yes that means if a value was originally a number on the server, it will now be a string when the boost is updated. Via my testing, this appears to be functionally equivalent. Likewise, if a value was an object previously, it will be serialize to "[object Object]". Again, an object is not valid anyway, so I think this is acceptable.